runtime/floats.c

```diff
@@ -461,11 +464,63 @@ CAMLprim value caml_exp_float(value f)
   return caml_copy_double(exp(Double_val(f)));
 }
 
+CAMLexport double caml_trunc(double x)
+{
+#ifdef HAS_C99_FLOAT_OPS
+  return trunc(x);
+#else
+  return (x >= 0.0)? floor(x) : ceil(x);
+#endif
+}
+
+CAMLprim value caml_trunc_float(value f)
+{
+  return caml_copy_double(caml_trunc(Double_val(f)));
+}
+
+CAMLexport double caml_round(double f)
+{
+#ifdef HAS_C99_FLOAT_OPS
+  return round(f);
+#else
+  union { uint64_t i; double d; } u, pred_one_half; /* predecessor of 0.5 */
```

Comment:
I don't feel competent to review this emulation of `round`, but I'm read to trust @Chris00 here.  If you have some reference to a description of this algorithm, it might be a good idea to insert it in the code.